pkg/defaults/components.go

```diff
@@ -26,42 +26,61 @@ var log = core.Log.WithName("defaults")
 
 func Setup(runtime runtime.Runtime) error {
 	if runtime.Config().Mode != config_core.Zone { // Don't run defaults in Zone (it's done in Global)
-		defaultsComponent := NewDefaultsComponent(runtime.Config().Defaults, runtime.Config().Mode, runtime.Config().Environment, runtime.ResourceManager(), runtime.ResourceStore())
-
-		zoneIngressSigningKeyManager := tokens.NewSigningKeyManager(runtime.ResourceManager(), zoneingress.ZoneIngressSigningKeyPrefix)
-		if err := runtime.Add(tokens.NewDefaultSigningKeyComponent(
-			zoneIngressSigningKeyManager,
-			log.WithValues("secretPrefix", zoneingress.ZoneIngressSigningKeyPrefix))); err != nil {
+		// todo(jakubdyszkiewicz) once this https://github.com/kumahq/kuma/issues/1001 is done. Wait for all the components to be ready.
+		tenantIds, err := runtime.Tenant().GetTenantIds(context.TODO())
+		if err != nil {
 			return err
 		}
+		var allErrors error
+		for _, tenantId := range tenantIds {
+			ctx := user.Ctx(context.WithValue(context.TODO(), runtime.Tenant().TenantContextKey(), tenantId), user.ControlPlane)
+			defaultsComponent := NewDefaultsComponent(ctx, runtime.Config().Defaults, runtime.Config().Mode, runtime.Config().Environment, runtime.ResourceManager(), runtime.ResourceStore())
+			if err := runtime.Add(defaultsComponent); err != nil {
+				allErrors = multierr.Append(allErrors, err)
+			}
 
-		zoneSigningKeyManager := tokens.NewSigningKeyManager(runtime.ResourceManager(), zone.SigningKeyPrefix)
-		if err := runtime.Add(tokens.NewDefaultSigningKeyComponent(
-			zoneSigningKeyManager,
-			log.WithValues("secretPrefix", zoneingress.ZoneIngressSigningKeyPrefix),
-		)); err != nil {
-			return err
+			zoneIngressSigningKeyManager := tokens.NewSigningKeyManager(runtime.ResourceManager(), zoneingress.ZoneIngressSigningKeyPrefix)
+			if err := runtime.Add(tokens.NewDefaultSigningKeyComponent(ctx, zoneIngressSigningKeyManager, log.WithValues("secretPrefix", zoneingress.ZoneIngressSigningKeyPrefix))); err != nil {
```

Comment:
I guess it's fine in `Setup` before the CP is started. If it had it would've triggered - https://github.com/kumahq/kuma/blob/305b749b1b9a81ab5fbbb425fd80d67092e41f3f/pkg/core/runtime/component/component.go#L95-L100

Reply:
> Maybe you should keep this as is ftm and fix this in a follow up issue to not keep blocking this work.

👍